*Loc e Iloc*
loc: permite pegar uma linha de acordo com indice dela. Ele dá erro caso nao encontre o indice. loc[indice_linha, indice_coluna]

iloc: exerga o dataframe como linhas e colunas, e se consegue acessar o calor atraves do numero do indice e da coluna, apenas a posicao importa. iloc[num_linha, num_coluna]

In [2]:
import pandas as pd
novo_prdodutos_df = pd.read_csv('Contoso - Cadastro Produtos.csv', sep=';')
display(novo_prdodutos_df.head())

Nome da Marca       Tipo Custo Unitario Preco Unitario  ID Produto  \
0       Contoso  Econômico          10,69          20,96         873   
1       Contoso  Econômico           6,63             13         879   
2       Contoso  Econômico           6,63             13         880   
3       Contoso  Econômico           6,63             13         881   
4       Contoso  Econômico           6,63             13         882   

                   Categoria  
0  Acessório para Computador  
1  Acessório para Computador  
2  Acessório para Computador  
3  Acessório para Computador  
4  Acessório para Computador

In [14]:
#metodo loc
print(novo_prdodutos_df.loc[0, 'Custo Unitario'])
#metodo iloc 
print(novo_prdodutos_df.iloc[0, 2])


10,69
10,69


Se eu quisesse alterar o preco do produto com Id 873, como faria?

In [16]:
novo_prdodutos_df.loc[0, 'Preco Unitario'] = 25
display(novo_prdodutos_df)

Nome da Marca       Tipo Custo Unitario Preco Unitario  ID Produto  \
0              Contoso  Econômico          10,69             25         873   
1              Contoso  Econômico           6,63             13         879   
2              Contoso  Econômico           6,63             13         880   
3              Contoso  Econômico           6,63             13         881   
4              Contoso  Econômico           6,63             13         882   
...                ...        ...            ...            ...         ...   
1685  Southridge Video    Regular          82,77         179,99        1602   
1686  Southridge Video    Regular          73,57         159,99        1606   
1687  Southridge Video    Regular          82,77         179,99        1607   
1688  Southridge Video    Regular          73,57         159,99        1611   
1689  Southridge Video    Regular          82,77         179,99        1612   

                      Categoria  
0     Acessório para Computador  
1     Acessório para Computador  
2     Acessório para Computador  
3     Acessório para Computador  
4     Acessório para Computador  
...                         ...  
1685                  Filme DVD  
1686                  Filme DVD  
1687                  Filme DVD  
1688                  Filme DVD  
1689                  Filme DVD  

[1690 rows x 6 columns]